packages/hash/api/src/graphql/resolvers/entityType/updateEntityType.ts

```diff
@@ -7,13 +9,30 @@ export const updateEntityType: Resolver<
   {},
   LoggedInGraphQLContext,
   MutationUpdateEntityTypeArgs
-> = async (_, { accountId, entityId, schema }, { dataSources, user }) => {
-  const entityType = await EntityType.updateSchema(dataSources.db, {
-    accountId,
-    updatedByAccountId: user.accountId,
-    createdByAccountId: user.entityId,
-    entityId,
-    schema,
+> = async (
+  _,
+  { accountId, entityId, schema },
+  { dataSources: { db }, user },
+) => {
+  const entityType = await EntityType.getEntityType(db, {
+    entityTypeId: entityId,
   });
-  return entityType.toGQLEntityType();
+
+  if (!entityType) {
+    throw new ApolloError(
+      `EntityType with entityId ${entityId} not found`,
+      "NOT_FOUND",
+    );
+  }
+
+  return await db.transaction(async (conn) =>
+    (
+      await entityType.update(conn, {
+        accountId,
+        updatedByAccountId: user.accountId,
+        createdByAccountId: user.entityId,
+        schema,
+      })
+    ).toGQLEntityType(),
+  );
```

Comment:
Is there a reasoning why only the update occurs in the transaction? Is it to reduce the size the transaction?

Reply:
No reason for this. I'll wrap it all in the transaction :)

Reply:
Okay, well I think it's equivalent so doesn't require changing - it's just not consistent with other resolvers where we do everything in one transaction (including the fetching).

Once we consistently lock entities that we are updating on retrieval, the fetch will need to occur in the transaction though.